qiskit/transpiler/passes/crosstalk_adaptive_schedule.py

```diff
@@ -0,0 +1,667 @@
+        for ginfo in backend_prop.gates:
+            if ginfo.gate == 'u1':
+                q_0 = ginfo.qubits[0]
+                for item in ginfo.parameters:
+                    if item.name == 'gate_error':
+                        if item.value == 1.0:
+                            self.bp_u1_err[q_0] = 0.999999
+                        else:
+                            self.bp_u1_err[q_0] = round(item.value, NUM_PREC)
+                    elif item.name == "gate_length":
+                        self.bp_u1_dur[q_0] = int(item.value)
+            elif ginfo.gate == 'u2':
+                q_0 = ginfo.qubits[0]
+                for item in ginfo.parameters:
+                    if item.name == 'gate_error':
+                        if item.value == 1.0:
+                            self.bp_u2_err[q_0] = 0.999999
+                        else:
+                            self.bp_u2_err[q_0] = round(item.value, NUM_PREC)
+                    elif item.name == "gate_length":
+                        self.bp_u2_dur[q_0] = int(item.value)
+            elif ginfo.gate == 'u3':
+                q_0 = ginfo.qubits[0]
+                for item in ginfo.parameters:
+                    if item.name == 'gate_error':
+                        if item.value == 1.0:
+                            self.bp_u3_err[q_0] = 0.999999
+                        else:
+                            self.bp_u3_err[q_0] = round(item.value, NUM_PREC)
+                    elif item.name == "gate_length":
+                        self.bp_u3_dur[q_0] = int(item.value)
+            elif ginfo.gate == 'cx':
+                q_0 = ginfo.qubits[0]
+                q_1 = ginfo.qubits[1]
+                r_0 = min(q_0, q_1)
+                r_1 = max(q_0, q_1)
+                for item in ginfo.parameters:
+                    if item.name == 'gate_error':
+                        if item.value == 1.0:
+                            self.bp_cx_err[(r_0, r_1)] = 0.999999
+                        else:
+                            self.bp_cx_err[(r_0, r_1)] = round(item.value, NUM_PREC)
+                    elif item.name == "gate_length":
+                        self.bp_cx_dur[(r_0, r_1)] = int(item.value)
+
+    def cx_tuple(self, gate):
+        """
+        Representation for two-qubit gate
+        """
+        physical_q_0 = gate.qargs[0].index
+        physical_q_1 = gate.qargs[1].index
+        r_0 = min(physical_q_0, physical_q_1)
+        r_1 = max(physical_q_0, physical_q_1)
+        return (r_0, r_1)
+
+    def singleq_tuple(self, gate):
+        """
+        Representation for single-qubit gate
+        """
+        physical_q_0 = gate.qargs[0].index
+        tup = (physical_q_0)
+        return tup
+
+    def gate_tuple(self, gate):
+        """
+        Representation for gate
+        """
+        if len(gate.qargs) == 2:
+            return self.cx_tuple(gate)
+        else:
+            return self.singleq_tuple(gate)
+
+    def assign_gate_id(self, dag):
```

Comment:
I use the gate ids to create Z3 variables and for symmetry breaking in the overlap constraints. I think we should keep the integer ids if that's ok. It helps because the Z3 vars created in `create_z3_vars` and the constraints in `basic_bounds`, `scheduling_constraints` etc. are easy to print and read for future debugging purposes. 
`if len(g_2.qargs) == 2 and self.gate_id[g_1] > self.gate_id[g_2]` is used to create only 1 overlap variable for a pair of gates.

Reply:
sounds good. i was just curious whether it's beneficial for the DAG class to also expose an integer id